core/chains/evm/log/broadcaster.go

```diff
@@ -350,23 +350,24 @@ func (b *broadcaster) eventLoop(chRawLogs <-chan types.Log, chErr <-chan error)
 	var needsResubscribe bool
 	debounceResubscribe := time.NewTicker(1 * time.Second)
 	defer debounceResubscribe.Stop()
+	var latestSeenLogBlockNumber int64
 
 	b.logger.Debug("Starting the event loop")
 	for {
 		select {
 		case rawLog := <-chRawLogs:
-
 			b.logger.Debugw("Received a log",
 				"blockNumber", rawLog.BlockNumber, "blockHash", rawLog.BlockHash, "address", rawLog.Address)
-
 			b.onNewLog(rawLog)
+			latestSeenLogBlockNumber = int64(rawLog.BlockNumber)
 
 		case <-b.newHeads.Notify():
 			b.onNewHeads()
 
 		case err := <-chErr:
-			// Note we'll get a message on this channel
-			// if the eth node terminates the connection.
+			// The eth node connection was terminated so we need to backfill after resubscribing.
+			b.backfillBlockNumber.SetValid(latestSeenLogBlockNumber)
```

Comment:
- [ ] One edge case left: if `latestSeenLogBlockNumber` is uninitialized, then I guess we can just skip and let it do the default depth? Or maybe we could init with something meaningful?